### DISCUSSION TITLE
Making use of `vim.ui.select()` in `register_template`

### DISCUSSION THREAD

**Discussion by pidgeon777** [stevearc/overseer.nvim#229] [Q&A]
Actually my code is the following:

```lua
see.register_template({
  name = "Template Name",
  builder = function()
    local task_name = ""

    vim.ui.select("Selected name: ", task_names, function(selected_val)
      if selected_val == nil then
        return
      else
        task_name = selected_val
      end
    end)

    return {
      name = task_name,
      ...
    }
  end,
})
```

The problem is that `vim.ui.select` is making use of a callback function which sets the value of `task_name` after the template has been built.

So, I would like to ask how to make use of `vim.ui.select` so that I can populate the template with the selected value, before returning it.

Maybe also a way to make `vim.ui.select` synchronous, so that it blocks the execution of the template builder until a value is selected?

**Comment by stevearc** [comment]
Something similar to this was discussed in #86, with a workaround. Unfortunately there's not a great way to do this at the moment because the `builder()` function is not async. Refactoring it to be async could be a little tricky (since we have to maintain backwards compatibility). I would like to generally refactor the interface for adding tasks/templates, but I barely have the bandwidth to address bug reports these days.

A couple other workarounds to try: `vim.fn.inputlist` is not a great interface but it _is_ synchronous. You could also try using `vim.wait` after the call to `vim.ui.select`. I _think_ that will just block the UI, but I'm only about 70% sure.
